frontend/src/scenes/billing/billingProductLogic.ts

```diff
@@ -109,6 +110,12 @@ export const billingProductLogic = kea<billingProductLogicType>([
                 setSurveyID: (_, { surveyID }) => surveyID,
             },
         ],
+        billingProductLoading: [
+            null as string | null,
+            {
+                setBillingProductLoading: (_, { productKey }) => productKey,
+            },
+        ],
```

Comment:
I feel like this should be getting reset somewhere when it's done loading? If the page refreshes then makes sense that it doesn't get manually reset, but the page doesn't always refresh.

Reply:
It probably makes more sense to move the call on the subscribe / add buttons into a loader so the loading state can be handled by kea.

Reply:
It can be the same as the deactivate loader currently on billingLogic.tsx

Reply:
These are all outgoing links, so that's why I did it this way. I only added these to buttons with `to` props. 

I could move into kea but it would just be a `window.location.href = ` and the loader wouldn't get reset anyways.

Reply:
Agree it probably makes sense to move these into kea as a fetch instead so I can do that now.

Reply:
I don't think it makes sense to move to kea since sometimes we will need to redirect to checkout but that won't be known at time of click, right?

Reply:
They're not always outgoing links, no (see `disableClientSideRouting` . If there's an existing subscription then the subscription just happens and the billing response is re-fetched, which triggers a re-render of the page (not a full refresh). It seems like it worked somehow in your demo video though so something is obviously happening, just somewhat concerning that it's not clear _what_ is happening.

Reply:
I'm not sure I understand what you're saying.

I've added this loading state to two buttons for subscribing to Products and Addons. In both cases it's hitting `/api/billing-v2/activation?products=`. These are URLs are not in the react application so to my understanding, all the `disableClientSideRouting` is doing is making sure it doesn't [push to the kea router](https://github.com/PostHog/posthog/blob/zach/billing-activation-deactivation-loading/frontend/src/lib/lemon-ui/Link/Link.tsx#L120-L129).

Given it's navigating to URLs outside of the application it will always trigger a full page refresh, that's not avoidable unless we move the logic to a fetch - which we can't do unless we change the activate endpoint 
or add logic on the client around seeing if they need to be redirected to Stripe or not.

Reply:
Oh hm ok I thought `disableClientSideRouting` didn't trigger a full page refresh but I'll take your word for it, I haven't actually dug into the code for this in a while 😄

Reply:
Ok, I'll test it a bit more but I think it's a good solution given they are all outgoing links.

Reply:
I must be missing something. Why does it matter that you don’t know if they’re going to be redirected to stripe on the client?

Reply:
Because we are redirecting `/api/billing-v2/activation?products=`, sometimes that will redirect to Stripe and back to the app (if they don't have a Stripe customer) and other times it will hit the endpoint and redirect back right away (if they do have a Stripe customer).

Reply:
Sure, I see that, but it’s not clear why kea wouldn’t be able to handle that.

Reply:
The original question was:
> I feel like this should be getting reset somewhere when it's done loading?

So I'm just saying that all these requests are requests to the server that will [return a redirect](https://github.com/PostHog/posthog/blob/zach/remove-activate-all-products-button/ee/api/billing.py#L118) so the loading doesn't need to be reset because there always be a page refresh.

Reply:
Got it, thanks.

Reply:
I can look into moving these into kea as a fetch that gets redirected as well, that's just a bit of a different change.

Reply:
I feel like keeping the redirect logic in the billing service is fine. It keeps the logic around when to interact with stripe in one place.

Reply:
I did just something like:
```
                activateProduct: async ({ productKey, redirect }: { productKey: string, redirect?: string }) => {
                    let path = 'api/billing-v2/activation?products=' + productKey
                    if (redirect) {
                        path += '&redirect=' + redirect
                    }
                    return api.getResponse(path)
                }
```
but the API returns a redirect to the billing repo to then the fetch goes there and gets a CORS issue. So it's definitely better to keep it as is.